fire/cli/niv/_udtræk_observationer.py

```diff
@@ -0,0 +1,453 @@
+    assert (
+        len(geometrifiler) <= 1
+    ), "Det er indtil videre forventet, at der kun bliver oplyst én geometrifil ad gangen."
+
+    assert (
+        len(ubrugelige) == 0
+    ), f"Følgende kriterier er hverken en korrekt-formuleret ident eller geometrifil: {ubrugelige!r}"
+
+    # Genvejsvariabel
+    db = fire.cli.firedb
+
+    fire.cli.print("Klargør identer", bold=True)
+    identer: List[str] = [klargør_ident_til_søgning(ident) for ident in identer]
+
+    fire.cli.print("Søg i databasen efter punkter til hver ident", fg="yellow")
+    punkter: List[Punkt] = [db.hent_punkt(ident) for ident in identer]
+
+    DVR90 = fire.cli.firedb.hent_srid("EPSG:5799")
+    OBSKLASSE = {
+        NivMetode.MGL: GeometriskKoteforskel,
+        NivMetode.MTL: TrigonometriskKoteforskel,
+    }
+
+    # Punkt-scenarium 1 - Buffer == 0
+    # Vi søger kun observationer med identerne som opstillingspunkter.
+    if buffer == 0:
+
+        fire.cli.print("Hent observationer for identerne", fg="yellow")
+        hent_observationer = partial(
+            db.hent_observationer_fra_opstillingspunkt,
+            tidfra=fra,
+            tidtil=til,
+            srid=DVR90,
+            kun_aktive=True,
+        )
+        hentere = [
+            partial(hent_observationer, observationsklasse=OBSKLASSE[metode])
+            for metode in metoder
+        ]
+        resultatsæt = set()
+        for hent in hentere:
+            for punkt in punkter:
+                resultatsæt |= set(hent(punkt))
+
+    # / Scenarium 1
+
+    # Punkt-scenarium 2: Buffer er angivet
+    # Vi søger observationer og punkter i nærheden af identernes placering.
+
+    else:
+
+        # Hent punkternes WGS84-koordinater:
+        # Geometri-koordinaterne er altid i WGS84.
+        koordinatsæt = [punkt.geometri.koordinater for punkt in punkter]
+
+        # Opbyg geometri for punkt-koordinater til søgning.
+        shapely_punkter = [geometry.Point(*koordinater) for koordinater in koordinatsæt]
+
+        # Lav den endelige søge-geometri ved at bruge den angivne buffer som
+        # radius i en forsimplet cirkel (polygon) omkring koordinaterne.
+        shapely_polygoner = [punkt.buffer(buffer) for punkt in shapely_punkter]
+
+        # Tilføj polygonerne for de enkelte identer til geometrier, der skal søges i nærheden af.
+        # Opret samtidig et geometri-objekt med hver søge-geometris Well-Known Text (WKT).
+        søge_geometrier = [Geometry(polygon.wkt) for polygon in shapely_polygoner]
+
+        fire.cli.print(
+            f"Hent observationer inden for {buffer:d} m af identerne.", fg="yellow"
+        )
+        hent_observationer = partial(
+            db.hent_observationer_naer_geometri,
+            afstand=0,
+            tidfra=fra,
+            tidtil=til,
+        )
+        hentere = [
+            partial(hent_observationer, observationsklasse=OBSKLASSE[metode])
+            for metode in metoder
+        ]
+        resultatsæt = set()
```

Comment:
Jeg tror den her kan hives ud af både if og else blokkene og defineres engang for alle omkring linje 317. Der kan spares en linje, men derudover ser det ud til at du kan komme i problemer hvis der kun søges i en geometrifil og ikke angives nogle identer.

Reply:
Det fejler ikke i det tilfælde, men jo, det kan sagtens forsimples yderligere.

Jeg retter til.

Reply:
Årsagen til, at det ikke fejler uden identer, er fordi resultatsæt blev oprettet, om der er identer på kommandolinien eller ej. Så fejlen kunne opståpm hvis vi gik vlæk fra at bruge else.

Det er rettet.